core/src/test/kotlin/fr/sncf/osrd/pathfinding/PathfindingTest.kt

```diff
@@ -63,13 +63,23 @@ fun checkPathfindingSuccess(
     assertEquals(expectedPathItemsPos, pathSuccess.pathItemPositions)
 
     if (expectedBlocks != null) {
-        assertEquals(expectedBlocks.map { "block.${md5(it)}" }, pathSuccess.blocks)
+        assertEquals(
+            expectedBlocks.map { "block.${md5(it)}" },
+            pathSuccess.path.blocks.map { it.id },
+        )
     }
     if (expectedRoutes != null) {
-        assertEquals(expectedRoutes, pathSuccess.routes)
+        assertEquals(expectedRoutes, pathSuccess.path.routes.map { it.id })
     }
     if (expectedTrackSectionRanges != null) {
-        assertEquals(expectedTrackSectionRanges, pathSuccess.trackSectionRanges)
+        assertEquals(expectedTrackSectionRanges.size, pathSuccess.path.trackSectionRanges.size)
+        for ((expected, actual) in
+            expectedTrackSectionRanges zip pathSuccess.path.trackSectionRanges) {
+            assertEquals(expected.trackSection, actual.trackSection)
+            assertEquals(expected.begin, actual.begin)
+            assertEquals(expected.end, actual.end)
+            assertEquals(expected.direction, actual.direction)
+        }
```

Comment:
A little more lines to changes (but straightforward), but it seems more natural to use the actual type in test.
So move `expectedTrackSectionRanges` to `TrackSectionRange` and:
```suggestion
        assertEquals(expectedTrackSectionRanges, pathSuccess.path.trackSectionRanges)
```

Reply:
Fixed in https://github.com/OpenRailAssociation/osrd/pull/13352/commits/9998a9ad1025e53263b9d8435210ab5ea576c5bd

(I'll soon rebase and merge the commit into the previous one)